### DISCUSSION TITLE
Which is the purpose of automatic Binding to Coordinator?

### DISCUSSION THREAD

**Discussion by emandtf** [Koenkk/zigbee2mqtt#20247] [Devices]
Why a new paired device is automatically Binded to Coordinator? What is its purpose?
I tried to Unbind all Device Clusters/Endpoints from Coordinator and that specific device seems works just normal....so I don't understand why some device has this automatic initial binding.....

**Comment by pixeldoc2000** [comment]
This depends on the implementation of the Device.

Sometime bindings are needed to get a device to send its status regularly, like Battery Level for example. If you unbind the endpoints of your device, you may have problems down the road.

There maybe also binds which may just "copied" from another device which are not necessary, but this not be the norm.

**Reply by emandtf** [reply]
Battery level is sent by "Reporting" tab in which I have few rows of configurations about all data that I want that device sends to Z2M periodically.
With an empty or "full" Bind tab section, all those data seems well Reported by respecting set timings...

So I would curious why 85% of my devices have an initial Bind to Coordinator row(s) if them seems works fine even if I manually Unbind all present lines...
Binding to Coordinator seems useless... 🤔

**Reply by pixeldoc2000** [reply]
A bind is required for some devices to be able to report its status.

If you feel the urge to mess with your devices / zigbee2mqtt config, I am not going to stop you. B**ut please don't ask for support here if your setup breaks afterward.**

I suggest you make yourself familiar with the z2m source code and workings, ZigBee Protocol and specific Implementation of device Firmware like Tuya. Then you maybe deduct if something like a bind is necessary or not and improve zigbee2mqtt like fixing device profiles.

**Reply by emandtf** [reply]
Just to know: **WHO** is asking for help?

I'm asking a legit question because who has decided to set an Initial binding to Coordinator for those devices that I own (45 in total) SEEMS to not have any knowledge too.
If those devices seems works WITHOUT any Binding to Coord, why a dev have to let them set for us?
It seems that it's developed without any knowledge about if that specific devices need or not a binding to Coordinator.

So until I'm not asking for help about binding, please don't invent something that I've not said or done.

**Comment by lux73** [comment]
Its always the same: anyone asks anything - another one gives an detailed answer with the hint that samething could broke and the thread Starter dont like this...

This is the one reason why skilled User dont want to help anymore...

Just my 2cent

**Reply by emandtf** [reply]
Nope.
If someone (me) is asking FOR INFO, he is not interested in phrases like "_if you change THAT or THAT OTHER thing...._" because he NEVER asked something like "_what will happen if I change X, Y or Z?_" but instead just "_which is the purpose of feature/option X?_".
That kind of "detailed" explanation is not useful as an ANSWER because who answered was writing something that doesn't concern the question.

Specifically I'm not interested to change Binding informations, but after many tries ALL my devices seems works like normal even if I click ALL "Unbind" buttons. So the question is legit: why the Developer choosed to initially create that Bind if it seems useless for THOSE devices? 🤷‍♂️
Maybe for someone is absolutely required (I'm not saying not), but for my (12 different types) seems useless.

**Comment by grafalex82** [comment]
> Why a new paired device is automatically Binded to Coordinator? What is its purpose?

My observations that there are 2 kinds of devices:
- ones send reports specifically to the coordinator.
- others send reports to those devices who are subscribed for reports

There are also 2 kind of message types - reports and commands. Reports are used to report a status (e.g. temperature, or brightness change). Commands are used by controller type devices (switches, knobs) to perform an action on other device (e.g light), for example turn it on, or change a color. But if you consider a smart button, out of the box it does not have any specific recepient who it can send commands to. That is why the default converter binds its commands to the coordinator, so that coordinator can convert it to an action.

**Reply by emandtf** [reply]
I've understood what you written, but my dubt remains the same.
Zigbee devices, by default, just sends their info/events to Coordinator (Binding a Switch to a Light directly it's not taked in count now), so there isn't any "action" in any case: a Smart Plug has to send its Meters to Coordinator so there isn't any "action" here.
All my Smart Plugs via Zigbee2mqtt have 4 clusters bound to Coordinator, but after a click on their "Unbind" button (and a successfully message from the device), all of them continues to work like usual.
So what you said has some logic...but if Unbinding all Clusters seems doesn't change anything: **why was initially done by Z2M?**
Just this....

**Reply by grafalex82** [reply]
I also have the same question that I am trying to clarify [here](https://github.com/Koenkk/zigbee2mqtt/discussions/20743) (from a little bit different angle).

Unfortunately I have not found any guidance like 'Smart switch shall behave this way, make these reports, and allow binding in this fashion". ZCL and ZLL specifications describe low level building blocks, but do not explain how this shall work all together. As I see different vendors treat specifications in their own way. For example a few devices I have:
- Xiaomi aqara (first series) send reports, but do not send commands, do not support binding
- IKEA buttons send commands, support binding
- Tuya Knob sends either reports or commands depending on mode selected
